sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -29,6 +30,7 @@
     final static int RetryIntervalInMS = 1000; //Once we detect failover wait for 1 second before retrying request.
     final static int MaxRetryCount = 120;
     private final static int MaxServiceUnavailableRetryCount = 1;
+    private final static int MAX_QUERYPLAN_ADDRESS_RETRY_COUNT = 2;
```

Comment:
NIT: I would prefer a name like "NonIOBoundOperationRetryCount" - that way we can reuse the constant.

Reply:
Is 'NonIO' meant for non-user-data?

Reply:
I think keeping it as `MAX_QUERY_PLAN_AND_ADDRESS_REFRESH_COUNT` makes more sense since it directly relates to just query plan and address refresh.  
If we need to add more constants, we can merge it afterwards, for now, let's keep it separate to have better readability of the code.